### PR TITLE
Switch from testcafe driving the functional test to saucectl

### PR DESCRIPTION
**Pull request by shammowla**
When we reach a max parallel test run on saucelabs, the tests fail. Switched to using saucectl to let saucelabs drive the test and properly queue the test runs and provide proper spec reports in the console instead of an error exit code. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: .github/workflows/prod.yaml [hunk 1]
```diff
@@ -57,34 +66,47 @@ jobs:
...
       - name: Install dependencies
         if: steps.npm-cache.outputs.cache-hit != 'true'
         run: npm ci
       - name: Build
         run: npm run test:functional:build:prod
-      - name: Run TestCafe Tests
-        run: npx testcafe -q -c 5 "saucelabs:Chrome@latest:macOS 11.00"
         env:
           ALLOY_PROD_VERSION: ${{ steps.package-version.outputs.current-version }}
+      - name: Suacelabs setup
```
Comment (carterworks) [inline .github/workflows/prod.yaml:86]: Typo: should be "Saucelabs"

File: .github/workflows/prod.yaml [hunk 2]
```diff
@@ -101,34 +123,47 @@ jobs:
...
       - name: Install dependencies
         if: steps.npm-cache.outputs.cache-hit != 'true'
         run: npm ci
       - name: Build
         run: npm run test:functional:build:prod
-      - name: Run TestCafe Tests
-        run: npx testcafe -q -c 5 'saucelabs:Safari@latest:macOS 11.00'
         env:
           ALLOY_PROD_VERSION: ${{ steps.package-version.outputs.current-version }}
+      - name: Suacelabs setup
```
Comment (carterworks) [inline .github/workflows/prod.yaml:143]: Typo: should be "Saucelabs"

File: .github/workflows/prod.yaml [hunk 3]
```diff
@@ -145,23 +180,43 @@ jobs:
...
       - name: Install dependencies
         if: steps.npm-cache.outputs.cache-hit != 'true'
         run: npm ci
       - name: Build
         run: npm run test:functional:build:prod
-      - name: Run TestCafe Tests
-        run: npx testcafe -q -c 5 'saucelabs:IE@latest:Windows 10'
         env:
           ALLOY_PROD_VERSION: ${{ steps.package-version.outputs.current-version }}
+      - name: Suacelabs setup
```
Comment (carterworks) [inline .github/workflows/prod.yaml:200]: Typo: should be "Saucelabs"

**Review by carterworks — CHANGES_REQUESTED** [review]
- Is there a way to run these actions locally (https://github.com/nektos/act maybe?)'
- Where is the `.sauce/safari.yml`?
- Is this the one that is failing for MacOS test runs?

**Comment by shammowla** [comment]
> 

- Running the workflow to test locally would've saved me so much time. Going test act to see if it works.
- The macOS test runs aren't working with the saucelabs docker image, my guess is the node_module transfer is too big and the VM idles. For Safari, I'm using the current workflow test setup

**Pull request merged**